pyquil/api/_qam.py

```diff
@@ -60,18 +61,24 @@ def load(self, executable):
         return self
 
     @_record_call
-    def write_memory(self, *, region_name: str, offset: int = 0, value=None):
+    def write_memory(self, *, region_name: str, offset: int = 0, value: Union[Sequence[int], Sequence[float]] = None):
```

Comment:
well this is embarrassing... third time's the charm for my attempt to get the type hint right!

since `value` defaults to `None`, it should actually be `Optional[Union[Sequence[int], Sequence[float]]]`

Reply:
Ah, my bad, should have caught that one - thanks!

Reply:
Ok, this is getting ridiculous, because I got it wrong again!  Now I remember why I didn't recommend adding a type hint in my original comment.

We also allow `value` to bare `int` or `float` in addition to a `Sequence` of such.

Reply:
So I *think* (starting to question my sanity now), the Real Deal Complete Type Hint (TM) should be `Optional[Union[int, float, Sequence[int], Sequence[float]]]`.

that is quite a mouthful